sympy/solvers/solvers.py

```diff
@@ -350,7 +350,11 @@ def check_assumptions(expr, against=None, **assumptions):
         if test is expected:
             continue
         elif test is not None:
-            return False
+            if get is True:
+                return False, key
```

Comment:
For your case one of the testcase, that can be written is : 

```
assert check_assumptions(-5, True, integer=True, positive=True) == (False, 'positive')

```
you must write your testcases such that all the changes you have done, is covered.